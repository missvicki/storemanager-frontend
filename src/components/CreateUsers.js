import React from 'react';
import { Modal, Message, Button, Container, Form } from 'semantic-ui-react';


const CreateUsers = (props) => {
    const {
        name,
        user_name,
        role,
        password,
        onSubmitHandler,
        onChangeHandler,
        onDismissHandler,
        loading,
        userError,
        userMessage,
        handleChange
    } = props

    const userErorMsg = (
        <Message
            negative
            error
            header='Error!!!'
            content={userError}
            onDismiss={onDismissHandler}
            style={{ width: "50%", margin: "auto", color: "red" }}
        />
    );

    const userMsg = (
        <Message
            positive
            success
            content={userMessage}
            onDismiss={onDismissHandler}
            header='Success!!!'
            style={{ width: "50%", margin: "auto", color: "green" }}
        />
    );

    const roles = [{
        key: 1,
        text: "admin",
        value: "amin",
    }, {
        key: 2,
        text: "attendant",
        value: "attendant",
    },]


    return (
        <div>
            <Container textAlign="right">
                <br />
                <Modal trigger={<Button basic color='teal'>Create Users</Button>}>
                    <Modal.Header>Create a new User</Modal.Header>
                    <Modal.Content>
                        <Modal.Description>
                            <Form onSubmit={onSubmitHandler} success error>
                                {userMessage ? userMsg : null}
                                {userError ? userErorMsg : null}
                                <br />
                                <Form.Field>
                                    <input
                                        type="text"
                                        placeholder="First name, Last Name"
                                        name="name"
                                        value={name}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Field>
                                    <input
                                        type="text"
                                        placeholder="Username"
                                        name="user_name"
                                        value={user_name}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Select
                                    placeholder='Select roles'
                                    options={roles}
                                    name="role"
                                    onChange={handleChange}
                                />
                                <Form.Field>
                                    <input
                                        type="password"
                                        placeholder="Password"
                                        name="password"
                                        value={password}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Button type="submit" style={{ background: "teal", color: "white" }} loading={loading}>Create User</Form.Button>
                            </Form>
                        </Modal.Description>
                    </Modal.Content>
                </Modal>

            </Container>
        </div>
    );
}

export default CreateUsers;
