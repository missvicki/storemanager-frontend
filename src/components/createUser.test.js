import React from 'react';
import { mount } from 'enzyme';

import CreateUsers from './CreateUsers';

describe('Create Users', () => {
    const onChangeHandler = jest.fn();
    const onDismissHandler = jest.fn();
    const onSubmitHandler = jest.fn();

    const userProps = {
        user_name: "",
        name: "",
        role: "",
        password: "",
        onDismissHandler: onDismissHandler,
        userError: "",
        userMessage: "",
        loading:false,
        onSubmitHandler: onSubmitHandler,
        onChangeHandler: onChangeHandler
    }
    const createUser = mount(<CreateUsers {...userProps}/>)
    
    it('should render the Create User Component', () => {
        expect(createUser).toMatchSnapshot();
    });
});
