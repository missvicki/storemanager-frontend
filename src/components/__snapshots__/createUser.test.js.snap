// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Users should render the Create User Component 1`] = `
<CreateUsers
  loading={false}
  name=""
  onChangeHandler={[MockFunction]}
  onDismissHandler={[MockFunction]}
  onSubmitHandler={[MockFunction]}
  password=""
  role=""
  userError=""
  userMessage=""
  user_name=""
>
  <div>
    <Container
      textAlign="right"
    >
      <div
        className="ui right aligned container"
      >
        <br />
        <Modal
          centered={true}
          closeOnDimmerClick={true}
          closeOnDocumentClick={false}
          dimmer={true}
          eventPool="Modal"
          trigger={
            <Button
              as="button"
              basic={true}
              color="teal"
            >
              Create Users
            </Button>
          }
        >
          <Portal
            closeOnDocumentClick={false}
            closeOnEscape={true}
            eventPool="Modal"
            mountNode={<body />}
            onClose={[Function]}
            onMount={[Function]}
            onOpen={[Function]}
            onUnmount={[Function]}
            openOnTriggerClick={true}
            trigger={
              <Button
                as="button"
                basic={true}
                color="teal"
              >
                Create Users
              </Button>
            }
          >
            <Ref
              innerRef={[Function]}
            >
              <RefFindNode
                innerRef={[Function]}
              >
                <Button
                  as="button"
                  basic={true}
                  color="teal"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui teal basic button"
                        >
                          Create Users
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui teal basic button"
                          >
                            Create Users
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui teal basic button"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        Create Users
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </RefFindNode>
            </Ref>
          </Portal>
        </Modal>
      </div>
    </Container>
  </div>
</CreateUsers>
`;
